configuration_test.go

```diff
@@ -0,0 +1,71 @@
+package raft
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestCheckConfiguration(t *testing.T) {
+	var configuration Configuration
+	if checkConfiguration(configuration) == nil {
+		t.Fatalf("empty configuration should be error")
+	}
+
+	configuration.Servers = append(configuration.Servers, Server{
+		Suffrage: Nonvoter,
+		GUID:     "guid0",
+		Address:  "addr0",
+	})
+	if checkConfiguration(configuration) == nil {
+		t.Fatalf("lack of voter should be error")
+	}
+
+	configuration.Servers = append(configuration.Servers, Server{
+		Suffrage: Voter,
+		GUID:     "guid1",
+		Address:  "addr1",
+	})
+	if err := checkConfiguration(configuration); err != nil {
+		t.Fatalf("should be OK: %v", err)
+	}
+
+	configuration.Servers[1].GUID = "guid0"
+	if checkConfiguration(configuration) == nil {
+		t.Fatalf("duplicate GUID should be error")
+	}
+	configuration.Servers[1].GUID = "guid1"
+
+	configuration.Servers[1].Address = "addr0"
+	if checkConfiguration(configuration) == nil {
+		t.Fatalf("duplicate address should be error")
+	}
```

Comment:
For these kinds of tests it's nice to use `Strings.Contains()` on `err.Error()` to make sure it's failing for the correct reason.